packages/iTwinUI-react/src/core/Select/SelectTag.tsx

```diff
@@ -0,0 +1,67 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+import cx from 'classnames';
+import React from 'react';
+import SvgCloseSmall from '@itwin/itwinui-icons-react/cjs/icons/CloseSmall';
+import { useTheme, CommonProps } from '../utils';
+import '@itwin/itwinui-css/css/inputs.css';
+
+export type SelectTagProps = {
+  /**
+   * Callback function that handles click on close icon.
+   */
+  onCloseClick?: React.MouseEventHandler;
+  /**
+   * Callback function that handles kayboard click on close icon.
+   */
+  onCloseKeyDown?: React.KeyboardEventHandler;
+  /**
+   * Show close icon.
+   */
+  isRemovable?: boolean;
+  /**
+   * Text inside the tag.
+   */
+  children: React.ReactNode;
+} & CommonProps;
+
+/**
+ * Tag for showing selected value in `Select`.
+ * @private
+ */
+export const SelectTag = (props: SelectTagProps) => {
+  const {
+    className,
+    children,
+    isRemovable,
+    onCloseClick,
+    onCloseKeyDown,
+    ...rest
+  } = props;
+  useTheme();
+
+  return (
+    <span
+      className={cx('iui-select-tag', className)}
+      onClick={(e) => e.stopPropagation()}
+      {...rest}
+    >
+      <span className='iui-select-tag-label'>{children}</span>
+      {isRemovable && (
+        <button
```

Comment:
I was testing on a screen reader and noticed two issues:
- the left/right arrow key navigation conflicts with the VoiceOver commands
- the "remove" buttons are not _connected_ with the select tags, so it's hard to tell what is getting deleted.

For the first issue, I think we might be missing some `role`, because in other components like Tabs, the left/right keys don't conflict.

For the second issue, I feel like the whole tag should be a focusable button so that it action stays "connected" with the item, it might help with the first issue to some extent as well.

Reply:
I have added roles, which made the keyboard nav work, but they are not the best (couldnt find any, which could make sense).
When added those, screen reader stopped reading the `aria-label` from the buttom, which is weird....
If I moved focus to tag, there need more work, starting from the styling (no focus style on tag) and finishing with the action, cause then it is not abvious that clicking on tag will remove it :/

Reply:
@mayank99 can you check it up? Either we proceed with this one and later could come back for improvements or please suggest, how to improve it now or we get rid of closing icons from tags and remove a bunch of code.. Just with the last one kinda a pitty, cause most of the stuff works fine :/

Reply:
Maybe let's remove it and create a separate issue for this. Because it would be much harder to change things later after we ship. We could fork the branch and keep it around (with a link in the issue) if the concern is that all this work would be "lost".

I was trying to research and came across this detailed article which includes real user testing, and they have different suggestions to what we are currently doing.
https://www.24a11y.com/2019/select-your-poison-part-2/

Reply:
I have gotten rid of close icon on tags and all the removal logic.

Reply:
Can you create an issue to improve this part later?

Reply:
#790

Reply:
Testing on VO again, the select-button part is definitely better but i noticed that there is no feedback when you select/deselect there is no feedback

https://user-images.githubusercontent.com/9084735/185974811-fa3fdf54-9ed3-4671-9292-165ca863c221.mov